lib/streamlit/util.py

```diff
@@ -126,6 +127,11 @@ def index_(iterable, x) -> int:
     """
 
     for i, value in enumerate(iterable):
+        # https://stackoverflow.com/questions/588004/is-floating-point-math-broken
+        # https://github.com/streamlit/streamlit/issues/4663
+        if isinstance(iterable, np.ndarray):
+            if abs(x-value) < ERROR_CONSTANT:
+                return True
```

Comment:
The type returned here doesn't make sense -- index_ is supposed to return the index that the element was found at (or throw an exception nif it doesn't exist)

Reply:
oops. thanks.